static/examples/batch/batch-decomposition-threads.scd

```diff
@@ -14,17 +14,20 @@ s.waitForBoot({
     var counter = 0;
 
     // Remove any files from the audioFiles collection so that only those with a .wav extension remain
-    var wav = audioFiles.files.removeAllSuchThat({ |file| file.extension == "wav" });
+    var wav = audioFiles.files.select({ |file| file.extension == "wav" });
 
+    var pool = Semaphore(4); // Create a Semaphore that allows 4 consecutive jobs
     // In the language iterate over all the files in the example audio files directory
     wav.do({ |file|
         fork ({
-            var src = Buffer.read(s, file.fullPath.asString); // Create a buffer with the source audio in a buffer called "src"
+            var src, h, p, result;
+            pool.wait; // We tell the Semaphore above that we are going to be doing some work and one of the 4 workers needs to WAIT.
```

Comment:
Comment isn't quite accurate, I think: we're (in effect) queueing up this portion of work via the semaphore. If there are 4 jobs already underway, then this will wait.

Reply:
Nuanced that bad boy